src/javascript/app_2/App/Components/Elements/SettingsDialog/settings_control.jsx

```diff
@@ -0,0 +1,33 @@
+import React        from 'react';
+import PropTypes    from 'prop-types';
+import ToggleButton from '../toggle_button.jsx';
+import { localize } from '../../../../../_common/localize';
+
+class SettingsControl extends React.Component {
+    render() {
+        return (
+            <div className='settings-row' onClick={this.props.toggle || this.props.onClick}>
+                <span>{localize(this.props.name)}</span>
+                {this.props.toggle ?
+                    <ToggleButton
+                        bool={this.props.to_toggle}
+                        style={this.props.style}
+                    />
+                    :
+                    this.props.children
+                }
+            </div>
+        );
+    }
+};
```

Comment:
suggestion to use functional component: `const SettingsControl = () => (..);` :D